### ISSUE TITLE
Error handling + Promises might be really broken

### ISSUE THREAD

**Issue by hayes** [othiym23/async-listener#41]
The Promise instrumentation wraps callbacks to Promise methods in generic async-listener callback wrappers which allow async-listener error listeners to handle errors and prevent them from being thrown properly.  Unfortunately this conflicts with Promise's default error handling, since the errors can be prevented from being thrown, the users code which may expect thrown exceptions to reject the promises may leave promises unresolved.

**Comment by taion** [comment]
What do you mean? The following test passes, for example. Although I'm not sure why `resolve` has an `after` count of 0.

```js
test('throw', function throwTest(t) {
  var listener = addListner();

  listener.currentName = 'resolve';
  Promise.resolve()
    .then(function () {
      listener.currentName = 'then';
      throw 12345;
    })
    .catch(function (error) {
      t.strictEqual(error, 12345);
      listener.currentName = 'catch';
    });

  listener.currentName = 'setTimeout';
  setTimeout(function() {
    t.deepEqual(listener.root, expected);
    t.end();
  });

  process.removeAsyncListener(listener.listener);

  var expected = {
    name: 'root',
    children: [
      {
        name: 'resolve',
        children: [
          {
            name: 'then',
            children: [
              {
                name: 'catch',
                children: [],
                before: 0,
                after: 0,
                error: 0
              }
            ],
            before: 1,
            after: 1,
            error: 0
          }
        ],
        before: 1,
        after: 0,
        error: 0
      },
      {
        name: 'setTimeout',
        children: [],
        before: 1,
        after: 0,
        error: 0
      }
    ],
    before: 0,
    after: 0,
    error: 0
  }
});
```

**Comment by hayes** [comment]
You'll need a listener that has an error listener that returns false

**Comment by taion** [comment]
I see - never mind then.

**Comment by hayes** [comment]
I have not tried to repo this yet, been to busy, may not be an issue, opened this more to remind myself to look into it

**Comment by hayes** [comment]
closing this as it is resolved by #47
